### PR TITLE
Add Baby spawning with spawn eggs.

### PR DESCRIPTION
**Pull request by aramperes**
Adds babies to Animals when clicking on a mob with the same type of Spawn Egg.

GlowCow, GlowPig and GlowSheep need to get their super#entityInteract method called in order to have babies.

### CODE REVIEW

**Comment by Paulomart** [comment]
There is currently no setParent method, right? If not this would be either way out of scope.

**Comment by mastercoms** [comment]
I think canceling spectator mode is outside of the scope of this PR. I think it would be better done inside our network handlers.

**Comment by aramperes** [comment]
I've added an [API ](https://github.com/GlowstonePlusPlus/Glowkit/pull/15)to Glowkit for baby parenting, since Bukkit did not handle this. Also, this would be used when we get to the Baby AI, to get what entity babies should follow.

File: src/main/java/net/glowstone/entity/passive/GlowSheep.java [hunk 1]
```diff
@@ -68,6 +68,7 @@ private byte getColorByte() {
 
     @Override
     public boolean entityInteract(GlowPlayer player, InteractEntityMessage message) {
+        super.entityInteract(player, message);
         if (player.getGameMode().equals(GameMode.SPECTATOR)) return false;
         if (player.getItemInHand() == null) return false;
         switch (player.getItemInHand().getType()) {
```
Comment (mastercoms) [inline src/main/java/net/glowstone/entity/passive/GlowSheep.java:77]: You can't use shears on a baby sheep.

**Comment by mastercoms** [comment]
Merged in https://github.com/GlowstoneMC/GlowstonePlusPlus/commit/9a11c8e42eb5a37afce83e447099a724690a4132. Thank you.

**Pull request closed without merging**